pkg/helm/chartproxy/proxy.go

```diff
@@ -99,10 +104,20 @@ func (p *proxy) IndexFile(onlyCompatible bool) (*repo.IndexFile, error) {
 					}
 				}
 				if len(entry) > 0 {
+					if overwrites != "" {
+						klog.Infof("%s overwrites repo %s", helmRepo.Name, overwrites)
+						delKeys = append(delKeys, entry[0].Name+"--"+overwrites)
```

Comment:
I think we need here `entry[i]`, not `entry[0]`?

Reply:
no, this is outside the loop, we want to remove duplicate at the key level not at the version level. Also see the comment from @baijum, that is why I'm using name and not key.